pkg/ipcache/types/types.go

```diff
@@ -63,7 +63,6 @@ func NewResourceID(kind ResourceKind, namespace, name string) ResourceID {
 
 // NodeIDHandler is responsible for the management of node identities.
 type NodeIDHandler interface {
```

Comment:
This IPCache type is no longer needed, is it? Can we remove it and remove it from the IPCache `Configuration` completely? Thanks

Reply:
The Mutual Auth feature still uses it:
```
$ git grep -P "ipCache.GetNodeI[PD]" pkg/auth/
pkg/auth/authmap_gc.go:                 nodeID, exists := r.ipCache.GetNodeID(addr.IP)
pkg/auth/manager.go:    nodeIP := a.ipCache.GetNodeIP(key.remoteNodeID)
```
I suspect that could be avoided by giving the auth package a reference to the node manager directly, but I'm a bit hesitant to make such changes to a feature I have very little context on. I've pinged Mutual Auth folks about this PR and Marco will review, so I'm hoping they can carry additional simplifications as follow ups.

Reply:
Yep, should be easy to get rid of this and depend directly on the nodemanager. Will come up with a follow up.

Reply:
Ah, I see. Thanks!

Reply:
@gandro @pchaigno see https://github.com/cilium/cilium/pull/27106